stubs/array.stub

```diff
@@ -0,0 +1,55 @@
+<?php
+
+// array_filter, needs to deal with mode, extension ?
+
+/**
+ * @template TIn of mixed
+ * @template TReturn of mixed
+ *
+ * @param array<TIn> $input
+ * @param callable(TReturn, TIn): TReturn $callback
+ * @param TReturn $initial
+ *
+ * @return TReturn
+ */
+function array_reduce(
+	array $input,
+	callable $callback,
+	$initial = null
+) {}
+
+/**
+ * @template TKey of array-key
+ * @template TIn of mixed
+ * @template TReturn of mixed
+ *
+ * @param null|callable(TIn): TReturn $callback
+ * @param array<TKey, TIn> $input1
+ * @param array<TKey, TIn> ...$args
+ *
+ * @return array<TKey, TReturn>
+ */
+function array_map(
```

Comment:
Looking at the php docs, this one is incorrect, and it may need an extension as well.

Reply:
How is it incorrect?

Reply:
Every extra array becomes another param of the callback. e.g.

```php
array_map(
  fn(string $one, string $two) => $one . $two,
  ['a', 'b', 'c'],
  ['d', 'e', 'f']
); // ['ad', 'be', 'cf']